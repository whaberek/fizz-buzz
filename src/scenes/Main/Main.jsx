import React, { useState } from 'react';
import styled from 'styled-components';
import { Formik, Form, Field, ErrorMessage } from 'formik';

import { Button, Input, Item, List } from 'components';

import { validate } from 'shared/validators';
import { colors, texts } from 'shared/constants';
import { modulo, checkDay } from 'shared/helpers';

const Container = styled.div`
  padding: 30px;
`;

const Main = () => {
  const [data, setData] = useState([]);

  const _submit = ({number}) => {
    let items = [];

    if (number) {
      for (let i = 1; i <= number; i++) {
        if (modulo(i, 15)) {
          items.push(
            <>
              <Item color={colors.blue}>{checkDay() ? texts.wizz : texts.fizz} </Item>
              <Item color={colors.green}>{checkDay() ? texts.wuzz : texts.buzz}</Item>
            </>
          );
        } else if (modulo(i, 3)) {
          items.push(<Item color={colors.blue}>{checkDay() ? texts.wizz : texts.fizz}</Item>);
        } else if (modulo(i, 5)) {
          items.push(<Item color={colors.green}>{checkDay() ? texts.wuzz : texts.buzz}</Item>);
        } else {
          items.push(<Item>{i}</Item>);
        }
      }

      setData(items);
    }
  };

  return (
    <Container>
      <Formik
        initialValues={{number: 1}}
        onSubmit={_submit}
        validate={({ number }) => validate(number)}
      >
        <Form>
          <ErrorMessage name="number" component="div" />
          <Field
            component={Input}
            type="number"
            name="number"
          />
          <Button type="submit" label={texts.submit} />
        </Form>
      </Formik>
      <List data={data} />
    </Container>
  );
};

export { Main };
