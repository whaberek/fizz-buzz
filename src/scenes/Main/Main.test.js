import React from 'react';
import { shallow } from 'enzyme';

import { Main } from './Main';

describe('Main', () => {
  it('should render layout', () => {
    const wrapper = shallow(<Main />);
    expect(wrapper).toMatchSnapshot();
  });
});
