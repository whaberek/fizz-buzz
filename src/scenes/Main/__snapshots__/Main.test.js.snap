// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render layout 1`] = `
<styled.div>
  <Formik
    enableReinitialize={false}
    initialValues={
      Object {
        "number": 1,
      }
    }
    isInitialValid={false}
    onSubmit={[Function]}
    validate={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
  >
    <Form>
      <FormikConnect(ErrorMessageImpl)
        component="div"
        name="number"
      />
      <FormikConnect(FieldInner)
        component={[Function]}
        name="number"
        type="number"
      />
      <Button
        label="Submit"
        type="submit"
      />
    </Form>
  </Formik>
  <List
    data={Array []}
  />
</styled.div>
`;
